src/components/ContributionButtonWithReminder.tsx

```diff
@@ -93,60 +106,60 @@ const RemindMeButton = ({ remindMeButtonText, onClick }: RemindMeButtonProps) =>
-    remindMeButtonText?: string;
-    remindMeConfirmationText?: string;
-    remindMeConfirmationHeaderText?: string;
+    reminderStage?: ReminderStage;
+    reminderOption?: string;
     trackClick: (buttonId: number) => void;
     hidePaymentIcons?: string;
 }
-type SectionState = 'DEFAULT' | 'REMINDER_CONFIRMED' | 'REMINDER_CONFIRMATION_CLOSED';
 
 export const ContributionButtonWithReminder = ({
     buttonText,
     buttonUrl,
-    remindMeButtonText,
-    remindMeConfirmationText,
-    remindMeConfirmationHeaderText,
+    reminderStage,
+    reminderOption,
     trackClick,
     hidePaymentIcons,
 }: ContributionButtonWithReminderProps): JSX.Element => {
-    const [sectionState, setSectionState] = useState<SectionState>('DEFAULT');
+    const { reminderCta, reminderPeriod, reminderLabel } = buildReminderFields();
+    const [reminderStatus, setReminderStatus] = useState<ReminderStatus>(ReminderStatus.Editing);
+    const [reminderConfirmationOpen, setReminderConfirmationOpen] = useState(false);
     const showPaymentIcons = hidePaymentIcons !== 'true';
 
-    if (sectionState === 'REMINDER_CONFIRMED') {
+    if (reminderConfirmationOpen) {
+        // Show instead of the ctas
         return (
             <ReminderCtaConfirmation
-                remindMeConfirmationText={remindMeConfirmationText as string}
-                remindMeConfirmationHeaderText={remindMeConfirmationHeaderText as string}
-                onClose={() => setSectionState('REMINDER_CONFIRMATION_CLOSED')}
+                success={reminderStatus === ReminderStatus.Completed}
+                label={reminderLabel}
+                onClose={() => setReminderConfirmationOpen(false)}
             />
         );
     }
 
-    if (sectionState === 'REMINDER_CONFIRMATION_CLOSED') {
-        return (
-            <div css={buttonWrapperStyles}>
-                <PrimaryButton
-                    buttonText={buttonText}
-                    buttonUrl={buttonUrl}
-                    trackClick={trackClick}
-                />
-
-                <PaymentIcons />
-            </div>
-        );
-    }
-
     return (
         <div css={buttonWrapperStyles}>
             <PrimaryButton buttonText={buttonText} buttonUrl={buttonUrl} trackClick={trackClick} />
 
-            {remindMeButtonText && remindMeConfirmationText && (
+            {reminderStage && !reminderConfirmationOpen && (
                 <RemindMeButton
-                    remindMeButtonText={remindMeButtonText}
+                    disabled={reminderStatus !== ReminderStatus.Editing}
+                    remindMeButtonText={reminderCta}
                     onClick={() => {
                         trackClick(REMIND_ME_BUTTON_INTERNAL_ID);
-                        setSectionState('REMINDER_CONFIRMED');
+
+                        setReminderStatus(ReminderStatus.Submitting);
+
+                        createReminder({
+                            reminderPeriod,
+                            email: 'TODO',
```

Comment:
Previously we didn't have to worry about knowing the user's email. Because they were logged in, we knew which Braze user they were, and the reminder email was triggered directly in Braze.

But now we need to pass the email to the reminders API. I think for signed in users it's available on the platform, I believe for SDC users it pre-populates the email input so perhaps we could use the same data? It looks like a `fetchEmail` prop is passed from the platform: https://github.com/guardian/dotcom-rendering/blob/f7fc75d7f7b9bb446fc2131ec46feef18a0f50e8/dotcom-rendering/src/components/StickyBottomBanner/ReaderRevenueBanner.tsx#L386 (this is for a banner, but same thing I think).